### PR TITLE
shard block syncing

### PR DESCRIPTION
**Pull request by Honglei-Cong**

### CODE REVIEW

File: consensus/solo/solo.go [hunk 1]
```diff
@@ -202,18 +198,20 @@ func (self *SoloService) broadCrossShardHashMsgs(blkNum uint32, shardMsgs []xsha
...
+	crossShardMsgs := &msg.CrossShardMsgs{}
 	for shardID, shardMsg := range shardMsgMap {
 		msgHash := xshard_types.GetShardCommonMsgsHash(shardMsg)
 		sig, err := signature.Sign(self.Account, msgHash[:])
 		if err != nil {
 			log.Errorf("sign cross shard msg failed, msg hash:%s, error: %s", msgHash.ToHexString(), err)
 			return
 		}
+		sigData := make(map[uint64][]byte)
+		sigData[self.shardID.ToUint64()] = sig
```
Comment (laizy) [inline consensus/solo/solo.go]: use `0` instead since the key is node index , not shard id.

Comment (xiemylogos) [reply consensus/solo/solo.go]: SigData map[uint64][]byte ,this map key is node index

Comment (xiemylogos) [reply consensus/solo/solo.go]: had  modify SigData type map[uint32][]byte

File: core/chainmgr/xshard/xshard_block.go [hunk 2]
```diff
@@ -201,60 +200,61 @@ func GetCrossShardTxs(lgr *ledger.Ledger, account *account.Account, FromShardID
...
 			ShardMsg: &types.CrossShardMsgInfo{
-				FromShardID: common.NewShardIDUnchecked(0),
+				FromShardID: FromShardID.ParentID(),
 				MsgHeight:   parentblkNum,
 			},
 			Tx: tx,
 		}
+		crossShardInfo := make([]*types.CrossShardTxInfos, 0)
 		crossShardInfo = append(crossShardInfo, shardTxInfo)
 		crossShardMapInfos[FromShardID.ToUint64()] = crossShardInfo
```
Comment (laizy) [inline core/chainmgr/xshard/xshard_block.go]: why `FromShardID` here?

Comment (xiemylogos) [reply core/chainmgr/xshard/xshard_block.go]: should FromShardID.ParentID()

**Review by laizy — APPROVED** [review]

**Pull request merged**